pkg/providers/instance/instance.go

```diff
@@ -377,7 +377,8 @@ func (p *DefaultProvider) getCapacityType(nodeClaim *corev1beta1.NodeClaim, inst
 	if requirements.Get(corev1beta1.CapacityTypeLabelKey).Has(corev1beta1.CapacityTypeSpot) {
 		for _, instanceType := range instanceTypes {
 			for _, offering := range instanceType.Offerings.Available() {
-				if requirements.Get(v1.LabelTopologyZone).Has(offering.Zone) && offering.CapacityType == corev1beta1.CapacityTypeSpot {
+				if requirements.Compatible(offering.Requirements, scheduling.AllowUndefinedWellKnownLabels) == nil &&
```

Comment:
Does it make sense to just create a set of requirements that only selects on spot here and then just check compatibility rather than having to specifically call out to select it?

Reply:
I was honestly fine leaving it like this since it's slightly more verbose to inject the requirement. Seems fine either way though.